### PR TITLE
9897: Logs errors encountered by block prefetch workers.

### PR DESCRIPTION
**Pull request by colin-nolan**

### CODE REVIEW

**Comment by tfmorris** [comment]
Ugh!

             except Exception:		              except Exception:
                  pass

Now there's an anti-pattern if there ever was one.

File: sdk/python/arvados/arvfile.py [hunk 1]
```diff
@@ -516,7 +517,7 @@ def _block_prefetch_worker(self):
                     return
                 self._keep.get(b)
             except Exception:
-                pass
+                _logger.error(traceback.format_exc())
```
Comment (colin-nolan) [inline sdk/python/arvados/arvfile.py]: I agree that knowing that a failure has occurred is a step in the right direction. However, that solution doesn't provide any insight into the nature and origin of the related fault. In addition, it obfuscates the location of that exception handler, making it difficult for someone not familiar with the codebase to know where to start.

A traceback provides much more useful information, including the location you suggest would be useful (albeit expressed without knowledge of the higher level context).

Comment (colin-nolan) [inline sdk/python/arvados/arvfile.py]: I believe the purpose of this Pokémon exception handler is to suppress failures from unknown rare errors. Therefore, the message should scarcely be seen by users of the SDK. This being the case, it seems reasonable for it to be intended for developers.

With regards to the extra import: the import statement could be moved in the `except` block?

Comment (jrandall) [inline sdk/python/arvados/arvfile.py]: apparently `_logger.exception()` does include a stack trace in the output

Comment (jrandall) [inline sdk/python/arvados/arvfile.py]: Maybe include "Warning:" in the exception message as well so that people know this is not an expected exception.

Comment (colin-nolan) [inline sdk/python/arvados/arvfile.py]: I don't believe that this is the case:
```
>>> _logger = logging.getLogger("test")
>>> def test():
...     _logger.exception("Some message")
... 
>>> test()
ERROR:test:Some message
None
```

Comment (tomclegg) [inline sdk/python/arvados/arvfile.py]: @colin-nolan "`_logger.exception()` does include a stack trace" is true when called from an exception handler (as in ["This method should only be called from an exception handler."](https://docs.python.org/2/library/logging.html#logging.Logger.exception) :)

```python
import logging
logging.basicConfig()
_logger = logging.getLogger()
try:
    nan = 1/0
except:
    _logger.exception("boom")
```

```
ERROR:root:boom
Traceback (most recent call last):
  File "boom.py", line 5, in <module>
    nan = 1/0
ZeroDivisionError: integer division or modulo by zero
```

Comment (colin-nolan) [inline sdk/python/arvados/arvfile.py]: Apologies; I stand corrected. I wasn't aware that this method's behaviour was context sensitive.

**Comment by tomclegg** [comment]
@colin-nolan thanks!

**Pull request merged**